File: daily-coding-challenge/2026/july/js/21_WordBlender#345.js
/*

Word Blender

Given two words, return a new word by combining the first half of the first word with the second half of the second word.
  For odd-length words, the first half is the shorter half.

Tests:
  blendWords("turtle", "toucan") should return "turcan".
  blendWords("chipmunk", "flamingo") should return "chipingo".
  blendWords("falcon", "pelican") should return "falican".
  blendWords("hyena", "iguana") should return "hyana".
  blendWords("scorpion", "gorilla") should return "scorilla".
  blendWords("platypus", "wolverine") should return "platerine".

*/

function blendWords(word1, word2) {
  let str1 = word1.slice(0, Math.floor(word1.length / 2));
  let str2 = word2.slice(Math.floor(word2.length / 2));
  return str1 + str2; // `${str1}${str2}`
}

console.log(blendWords("turtle", "toucan")); // "turcan".
console.log(blendWords("chipmunk", "flamingo")); // "chipingo".
console.log(blendWords("falcon", "pelican")); // "falican".
console.log(blendWords("hyena", "iguana")); // "hyana".
console.log(blendWords("scorpion", "gorilla")); // "scorilla".
console.log(blendWords("platypus", "wolverine")); // "platerine".

File: daily-coding-challenge/2026/july/python/21_WordBlender#345.py
"""

Word Blender

Given two words, return a new word by combining the first half of the first word with the second half of the second word.
  For odd-length words, the first half is the shorter half.

Tests:
  blend_words("turtle", "toucan") should return "turcan".
  blend_words("chipmunk", "flamingo") should return "chipingo".
  blend_words("falcon", "pelican") should return "falican".
  blend_words("hyena", "iguana") should return "hyana".
  blend_words("scorpion", "gorilla") should return "scorilla".
  blend_words("platypus", "wolverine") should return "platerine".

"""

import math

def blend_words(word1, word2):
    str1 = word1[:math.floor(len(word1) / 2)]
    str2 = word2[math.floor(len(word2) / 2):]
    return str1 + str2 # f"{str1}{str2}"

print(blend_words("turtle", "toucan"))# "turcan".
print(blend_words("chipmunk", "flamingo"))# "chipingo".
print(blend_words("falcon", "pelican"))# "falican".
print(blend_words("hyena", "iguana"))# "hyana".
print(blend_words("scorpion", "gorilla"))# "scorilla".
print(blend_words("platypus", "wolverine"))# "platerine".
